web/impact/impact/tests/test_cancel_office_hour_session_view.py

```diff
@@ -0,0 +1,185 @@
+from pytz import timezone
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+
+from django.core import mail
+from django.urls import reverse
+
+from accelerator.models import UserRole
+from accelerator.tests.contexts import UserRoleContext
+from accelerator.tests.utils import days_from_now
+
+from impact.tests.api_test_case import APITestCase
+from impact.tests.factories import MentorProgramOfficeHourFactory
+from impact.v1.views.cancel_office_hour_session_view import (
+    DEFAULT_TIMEZONE,
+    MENTOR_NOTIFICATION,
+    PERMISSION_DENIED,
+    STAFF_NOTIFICATION,
+    CancelOfficeHourSessionView,
+    MentorProgramOfficeHour,
+)
+
+
+class TestCancelOfficeHourSession(APITestCase):
+    url = reverse(CancelOfficeHourSessionView.view_name)
+
+    def test_mentor_can_cancel_their_own_unreserved_office_hour(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        office_hour = MentorProgramOfficeHourFactory(
+            mentor=mentor, finalist=None
+        )
+        with self.login(email=mentor.email):
```

Comment:
the code from 33 to 36 is repeated many times. Should be encapsulated in a function like `self.post_cancellation(user, office_hour)`, which would return the response for the tests that need it.